pkg/operator/controller/credentials-secret/credentials_secret.go

```diff
@@ -124,11 +128,27 @@ func desiredCredentialsSecret(sourceSecret *corev1.Secret, destName types.Namesp
 		case operatorv1alpha1.ProviderTypeAWS:
 			secret.Data = sourceSecret.Data
 		}
-	} else {
-		secret.Data = sourceSecret.Data
+		return secret, nil
 	}
 
-	return secret
+	if extDNS.Spec.Provider.Type == operatorv1alpha1.ProviderTypeAWS {
+		secret.Data = make(map[string][]byte)
+		switch {
+		case len(sourceSecret.Data["credentials"]) > 0:
+			secret.Data["credentials"] = sourceSecret.Data["credentials"]
+		case len(sourceSecret.Data["aws_access_key_id"]) > 0 && len(sourceSecret.Data["aws_secret_access_key"]) > 0:
+			secret.Data["credentials"] = newConfigForStaticCreds(
+				string(sourceSecret.Data["aws_access_key_id"]),
+				string(sourceSecret.Data["aws_secret_access_key"]),
+			)
+		default:
+			return nil, fmt.Errorf("invalid secret for aws credentials")
```

Comment:
Since we check the contents of the secret now, it makes sense to do it for other providers too.
This can be done in a followup PR.